### PR TITLE
feat(PL-2250): add --auto-merge flag for joy release promote

### PR DESCRIPTION
**Pull request by davidmdm**

### CODE REVIEW

File: internal/release/promote/interactive_prompt_provider.go [hunk 1]
```diff
@@ -182,6 +178,11 @@ func (i *InteractivePromptProvider) ConfirmCreatingPromotionPullRequest() (bool,
 	return selectedAction == actions[0], nil
 }
 
+func (*InteractivePromptProvider) ConfirmAutoMergePullRequest() (answer bool, err error) {
+	err = survey.AskOne(&survey.Confirm{Message: "Do you wish to auto-merge the resulting PR?"}, &answer)
```
Comment (j-martin) [inline internal/release/promote/interactive_prompt_provider.go:182]: Do we ever show a diff before creating the PR?

Comment (davidmdm) [reply internal/release/promote/interactive_prompt_provider.go:182]: Yes the diff is shown just before:
image

Comment (j-martin) [reply internal/release/promote/interactive_prompt_provider.go:182]: Ah... yes...

File: internal/release/promote/promotion.go [hunk 2]
```diff
@@ -129,10 +131,22 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
 		return "", nil
 	}
 
-	prURL, err := p.perform(list)
+	if opts.TargetEnv.Spec.Promotion.FromPullRequests && !opts.AutoMerge {
```
Comment (silphid) [inline internal/release/promote/promotion.go:134]: I don't think you lend the proper semantic to `FromPullRequests` property. It should probably be properly commented and documented to make things clearer, but what it really controls is whether `joy pr promote` will consider that environment in the user-prompted list as a potential candidate for promoting a "source code" PR. We will need something else to configure `auto-merge`. Maybe a new `Environment.spec.promotion.autoMergeable` or `.allowAutoMerge` property?

**Review by silphid — CHANGES_REQUESTED** [review]
See my comment but I don't think you interpret the meaning of `TargetEnv.Spec.Promotion.FromPullRequests` correctly.

**Review by silphid — APPROVED** [review]

**Pull request merged**